bson/bsonrw/extjson_parser.go

```diff
@@ -273,6 +283,61 @@ func (ejp *extJSONParser) readValue(t bsontype.Type) (*extJSONValue, error) {
 
 		ejp.advanceState()
 		if t == bsontype.Binary && ejp.s == jpsSawValue {
+			// convert relaxed $uuid format
+			if ejp.relaxedUUID {
+				uuid, err := ejp.v.parseSymbol()
+				if err != nil {
+					return nil, err
+				}
+
+				// RFC 4122 defines the length of a UUID as 36 and the hyphens in a UUID as appearing
+				// in the 8th, 13th, 18th, and 23rd characters.
+				//
+				// See https://tools.ietf.org/html/rfc4122#section-3
+				valid := len(uuid) == 36 &&
+					string(uuid[8]) == "-" &&
+					string(uuid[13]) == "-" &&
+					string(uuid[18]) == "-" &&
+					string(uuid[23]) == "-"
+				if !valid {
+					return nil, fmt.Errorf("$uuid value does not follow RFC 4122 format regarding length and hyphens")
+				}
+
+				// remove hyphens
+				uuidNoHyphens := strings.Replace(uuid, "-", "", -1)
```

Comment:
If the UUID is malformed and has an even number of extra hypens, we may silently allow the value. E.g. `73ffd264-44b3-4c69-90e8-e7d1dfc035--`

Can we account for that, and add a test for that scenario?

Reply:
Added a new test for this within `bson_corpus_spec_test#TestRelaxedUUIDValidation()`. Had to account for extra hyphens with a new check for length after removing the hyphens.